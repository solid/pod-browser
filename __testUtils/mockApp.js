/**
 * Copyright 2020 Inrupt Inc.
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal in
 * the Software without restriction, including without limitation the rights to use,
 * copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
 * Software, and to permit persons to whom the Software is furnished to do so,
 * subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in
 * all copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
 * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
 * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
 * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
 * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
 * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 */

import {
  addStringNoLocale,
  addUrl,
  createSolidDataset,
  createThing,
  setStringNoLocale,
  setThing,
  setUrl,
} from "@inrupt/solid-client";
import { vcard, rdf, schema } from "rdf-namespaces";
import { chain } from "../src/solidClientHelpers/utils";
import { vcardExtras } from "../src/addressBook";

export const oidcPrefix = "http://www.w3.org/ns/solid/oidc#";
export const TOS_PREDICATE = `${oidcPrefix}tos_uri`;
export const POLICY_PREDICATE = `${oidcPrefix}policy_uri`;
export const CONTACTS_PREDICATE = `${oidcPrefix}contacts`;
export const LOGO_PREDICATE = `${oidcPrefix}logo_uri`;

export const APP_WEBID = "https://mockappurl.com";
export const APP_POLICY_URL = "https://mockappurl.com/privacy-policy";
export const APP_TOS_URL = "https://mockappurl.com/terms-of-service";

const contactThing = chain(
  createThing({ url: "https://mockappurl.com#contacts" }),
  (t) => addStringNoLocale(t, vcard.fn, "Example Contact"),
  (t) => addStringNoLocale(t, vcard.hasTelephone, "555-5555"),
  (t) => addStringNoLocale(t, vcard.hasEmail, "example@email.com"),
  (t) => addUrl(t, vcard.url, "https://examplewebsite.com")
);

const dataset = setThing(createSolidDataset(), contactThing);

export function mockApp() {
  return chain(
    createThing({ url: APP_WEBID }),
    (t) => addStringNoLocale(t, vcard.fn, "Mock App"),
    (t) => addUrl(t, vcardExtras("WebId"), APP_WEBID),
    (t) => addUrl(t, rdf.type, schema.SoftwareApplication), // just guessing what the type might be for apps
    (t) => addUrl(t, TOS_PREDICATE, APP_TOS_URL),
    (t) => addUrl(t, POLICY_PREDICATE, APP_POLICY_URL),
    (t) => addUrl(t, CONTACTS_PREDICATE, "https://mockappurl.com#contacts")
  );
}

export function mockAppDataset() {
  return setThing(dataset, mockApp());
}
