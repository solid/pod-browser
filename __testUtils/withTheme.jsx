/**
 * Copyright 2020 Inrupt Inc.
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal in
 * the Software without restriction, including without limitation the rights to use,
 * copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
 * Software, and to permit persons to whom the Software is furnished to do so,
 * subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in
 * all copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
 * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
 * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
 * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
 * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
 * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 */

import React from "react";
import { render } from "@testing-library/react";
import T from "prop-types";
import { StylesProvider, ThemeProvider } from "@inrupt/prism-react-components";
import { FeatureProvider } from "../src/contexts/featureFlagsContext";
import defaultTheme from "../src/theme";

export default function WithTheme(props) {
  const { theme, children } = props;
  return (
    <StylesProvider generateClassName={(rule) => `PodBrowser-${rule.key}`}>
      <ThemeProvider theme={theme}>{children}</ThemeProvider>
    </StylesProvider>
  );
}

WithTheme.propTypes = {
  // eslint-disable-next-line react/forbid-prop-types
  theme: T.object,
  children: T.node.isRequired,
};

WithTheme.defaultProps = {
  theme: defaultTheme,
};

export const renderWithTheme = (children, theme = defaultTheme) => {
  return render(
    <FeatureProvider>
      <WithTheme theme={theme}>{children}</WithTheme>
    </FeatureProvider>
  );
};
