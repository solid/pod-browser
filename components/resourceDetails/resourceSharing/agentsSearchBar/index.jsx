/**
 * Copyright 2020 Inrupt Inc.
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal in
 * the Software without restriction, including without limitation the rights to use,
 * copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
 * Software, and to permit persons to whom the Software is furnished to do so,
 * subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in
 * all copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
 * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
 * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
 * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
 * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
 * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 */

import React from "react";
import PropTypes from "prop-types";
import { useBem } from "@solid/lit-prism-patterns";
import clsx from "clsx";
import { makeStyles } from "@material-ui/styles";
import { IconButton, InputBase, createStyles } from "@material-ui/core";
import styles from "./styles";

export const TESTCAFE_ID_SEARCH_INPUT = "search-input";

const useStyles = makeStyles((theme) => createStyles(styles(theme)));

export default function AgentsSearchBar({ handleFilterChange }) {
  const bem = useBem(useStyles());

  const classes = useStyles();
  return (
    <div className={classes.searchBoxContainer}>
      <IconButton type="submit" aria-label="search">
        <i
          className={clsx(bem("icon-search"), bem("icon"), classes.iconSearch)}
          aria-label="search"
        />
      </IconButton>
      <InputBase
        classes={{ root: classes.searchInput }}
        placeholder="Search by name or WebId"
        inputProps={{
          "data-testid": TESTCAFE_ID_SEARCH_INPUT,
          "aria-label": "Search by name or WebId",
          onChange: handleFilterChange,
        }}
      />
    </div>
  );
}

AgentsSearchBar.propTypes = {
  handleFilterChange: PropTypes.func,
};

AgentsSearchBar.defaultProps = {
  handleFilterChange: () => {},
};
