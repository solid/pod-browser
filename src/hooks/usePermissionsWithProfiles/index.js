/**
 * Copyright 2020 Inrupt Inc.
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal in
 * the Software without restriction, including without limitation the rights to use,
 * copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
 * Software, and to permit persons to whom the Software is furnished to do so,
 * subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in
 * all copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
 * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
 * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
 * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
 * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
 * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 */

import { useSession } from "@inrupt/solid-ui-react";
import { useState, useEffect } from "react";
import { AUTHENTICATED_AGENT_PREDICATE } from "../../models/contact/authenticated";
import { PUBLIC_AGENT_PREDICATE } from "../../models/contact/public";
import { fetchProfile } from "../../solidClientHelpers/profile";

export default function usePermissionsWithProfiles(permissions) {
  const [permissionsWithProfiles, setPermissionsWithProfiles] = useState([]);
  const { fetch } = useSession();

  useEffect(() => {
    if (!permissions) return;
    Promise.all(
      permissions.map(async (p) => {
        let profile;
        let profileError;
        if (
          p.webId === PUBLIC_AGENT_PREDICATE ||
          p.webId === AUTHENTICATED_AGENT_PREDICATE
        ) {
          return p;
        }
        try {
          profile = await fetchProfile(p.webId, fetch);
        } catch (error) {
          profileError = error;
        }
        return {
          ...p,
          profile,
          profileError,
        };
      })
    ).then((completed) => setPermissionsWithProfiles(completed));
  }, [permissions, fetch]);
  return { permissionsWithProfiles };
}
