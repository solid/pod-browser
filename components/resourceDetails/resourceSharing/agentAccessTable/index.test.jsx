/**
 * Copyright 2020 Inrupt Inc.
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal in
 * the Software without restriction, including without limitation the rights to use,
 * copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
 * Software, and to permit persons to whom the Software is furnished to do so,
 * subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in
 * all copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
 * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
 * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
 * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
 * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
 * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 */

import React from "react";
import { waitFor } from "@testing-library/dom";
import userEvent from "@testing-library/user-event";
import { renderWithTheme } from "../../../../__testUtils/withTheme";
import AgentAccessTable, { TESTCAFE_ID_AGENT_ACCESS_TABLE } from "./index";
import { createAccessMap } from "../../../../src/solidClientHelpers/permissions";
import usePolicyPermissions from "../../../../src/hooks/usePolicyPermissions";
import usePermissionsWithProfiles from "../../../../src/hooks/usePermissionsWithProfiles";
import { TESTCAFE_ID_SEARCH_INPUT } from "../agentsSearchBar";
// import {
//   TESTCAFE_ID_TAB_PEOPLE,
//   TESTCAFE_ID_TAB_GROUPS,
// } from "../agentsTableTabs";
import {
  PUBLIC_AGENT_PREDICATE,
  PUBLIC_AGENT_TYPE,
} from "../../../../src/models/contact/public";
import {
  AUTHENTICATED_AGENT_PREDICATE,
  AUTHENTICATED_AGENT_TYPE,
} from "../../../../src/models/contact/authenticated";

jest.mock("../../../../src/hooks/usePolicyPermissions");
const mockedUsePolicyPermissions = usePolicyPermissions;

jest.mock("../../../../src/hooks/usePermissionsWithProfiles");
const mockedUsePermissionsWithProfiles = usePermissionsWithProfiles;

const profile1 = {
  avatar: null,
  webId: "https://example1.com/profile/card#me",
  name: "Example A",
  types: ["https://schema.org/Person"],
};

const profile2 = {
  avatar: null,
  webId: "https://example2.com/profile/card#me",
  name: "Example B",
  types: ["https://schema.org/Person"],
};

const profile3 = {
  avatar: null,
  webId: "https://example3.com/profile/card#me",
  name: "Example C",
  types: ["https://schema.org/Person"],
};

const profile4 = {
  avatar: null,
  webId: "https://example4.com/profile/card#me",
  name: "Example D",
  types: ["https://schema.org/Person"],
};

const profiles = [profile1, profile2, profile3, profile4];

const publicPermission = {
  acl: createAccessMap(true, true, false, false),
  webId: PUBLIC_AGENT_PREDICATE,
  type: PUBLIC_AGENT_TYPE,
};

const authenticatedPermission = {
  acl: createAccessMap(true, true, false, false),
  webId: AUTHENTICATED_AGENT_PREDICATE,
  type: AUTHENTICATED_AGENT_TYPE,
};

const permissions = [
  {
    acl: createAccessMap(true, true, false, false),
    webId: "https://example1.com/profile/card#me",
    type: "agent",
  },
  {
    acl: createAccessMap(true, true, false, false),
    webId: "https://example2.com/profile/card#me",
    type: "agent",
  },
  {
    acl: createAccessMap(true, true, false, false),
    webId: "https://example3.com/profile/card#me",
    type: "agent",
  },
  {
    acl: createAccessMap(true, true, false, false),
    webId: "https://example4.com/profile/card#me",
    type: "agent",
  },
];

const permissionsWithProfiles = permissions.map((p, i) => {
  return {
    ...p,
    profile: profiles[i],
  };
});

const permissionsWithProfilesMixedTypes = [
  {
    acl: createAccessMap(true, true, false, false),
    webId: "https://example1.com/profile/card#me",
    type: "agent",
    profile: profile1,
  },
  {
    acl: createAccessMap(true, true, false, false),
    webId: "https://example4.com/profile/card#me",
    type: "agent",
    profile: {
      avatar: null,
      name: "Not a person",
      types: ["Something else"],
    },
  },
];
describe("AgentAccessTable with agents", () => {
  beforeEach(() => {
    mockedUsePolicyPermissions.mockReturnValue({
      data: permissions,
      mutate: jest.fn(),
    });
    mockedUsePermissionsWithProfiles.mockReturnValue({
      permissionsWithProfiles,
    });
  });
  it("renders a list of permissions", async () => {
    const type = "editors";
    const { asFragment, queryAllByRole } = renderWithTheme(
      <AgentAccessTable type={type} />
    );

    expect(queryAllByRole("cell")).toHaveLength(3);
    expect(asFragment()).toMatchSnapshot();
  });
  it("shows all permissions when clicking 'show all' button", async () => {
    mockedUsePolicyPermissions.mockReturnValue({
      data: permissions,
    });
    mockedUsePermissionsWithProfiles.mockReturnValue({
      permissionsWithProfiles,
    });

    const type = "editors";
    const { queryAllByRole, getByTestId } = renderWithTheme(
      <AgentAccessTable type={type} />
    );
    const button = getByTestId("show-all-button");
    expect(queryAllByRole("cell")).toHaveLength(3);
    userEvent.click(button);
    expect(queryAllByRole("cell")).toHaveLength(4);
  });
  it("agents appear in alphabetical order with public and authenticated first", async () => {
    mockedUsePolicyPermissions.mockReturnValue({
      data: permissions,
    });
    mockedUsePermissionsWithProfiles.mockReturnValue({
      permissionsWithProfiles: [
        publicPermission,
        authenticatedPermission,
        ...permissionsWithProfiles,
      ],
    });

    const type = "editors";
    const { queryAllByRole, getByTestId } = renderWithTheme(
      <AgentAccessTable type={type} />
    );
    const button = getByTestId("show-all-button");
    expect(queryAllByRole("cell")).toHaveLength(3);
    userEvent.click(button);
    const cells = queryAllByRole("cell");
    expect(cells).toHaveLength(6);
    expect(cells[0]).toHaveTextContent("Anyone");
    expect(cells[1]).toHaveTextContent("Anyone signed in");
    expect(cells[2]).toHaveTextContent("Example A");
    expect(cells[3]).toHaveTextContent("Example B");
    expect(cells[4]).toHaveTextContent("Example C");
    expect(cells[5]).toHaveTextContent("Example D");
  });
  it("shows first 3 permissions by default and when clicking the 'hide' button", async () => {
    mockedUsePolicyPermissions.mockReturnValue({
      data: permissions,
    });
    mockedUsePermissionsWithProfiles.mockReturnValue({
      permissionsWithProfiles,
    });
    const type = "editors";
    const { queryAllByRole, getByTestId } = renderWithTheme(
      <AgentAccessTable type={type} permissions={permissions} />
    );
    const showAllButton = getByTestId("show-all-button");
    expect(queryAllByRole("cell")).toHaveLength(3);
    userEvent.click(showAllButton);
    expect(queryAllByRole("cell")).toHaveLength(4);
    const hideButton = getByTestId("hide-button");
    userEvent.click(hideButton);
    expect(queryAllByRole("cell")).toHaveLength(3);
  });
  it("renders a search box which filters by name or webId", () => {
    mockedUsePolicyPermissions.mockReturnValue({
      data: permissions,
    });
    mockedUsePermissionsWithProfiles.mockReturnValue({
      permissionsWithProfiles,
    });

    const type = "editors";
    const { getByTestId, queryByText } = renderWithTheme(
      <AgentAccessTable type={type} permissions={permissions} />
    );
    waitFor(() => {
      expect(queryByText("Example 2")).not.toBeNull();
      const searchInput = getByTestId(TESTCAFE_ID_SEARCH_INPUT);
      userEvent.type(searchInput, "2");
      expect(queryByText("Example 4")).toBeNull();
      expect(queryByText("Example 2")).not.toBeNull();
    });
  });
  // TODO: tabs have slightly changed so these tests need to be updated when tabs are restores
  it.skip("renders a set of tabs which filter by Group type", () => {
    const permissionsWithTypes = [
      {
        type: "agent",
        acl: createAccessMap(true, true, false, false),
        webId: "https://example1.com/profile/card#me",
      },
      {
        type: "group",
        acl: createAccessMap(true, true, false, false),
        webId: "https://example-group.com/profile/card#me",
      },
    ];

    mockedUsePolicyPermissions.mockReturnValue({
      data: permissionsWithTypes,
      mutate: jest.fn(),
    });
    mockedUsePermissionsWithProfiles.mockReturnValue({
      permissionsWithProfiles: permissionsWithProfilesMixedTypes,
    });
    const type = "editors";
    const { getByTestId, queryByText } = renderWithTheme(
      <AgentAccessTable type={type} />
    );
    waitFor(() => {
      // FIXME: change these test ids
      const tabPeople = getByTestId("tab-people");
      const tabGroups = getByTestId("tab-groups");
      userEvent.click(tabPeople);
      expect(queryByText("Example 1")).not.toBeNull();
      expect(queryByText("Not a person")).toBeNull();
      userEvent.click(tabGroups);
      expect(queryByText("No groups found")).not.toBeNull();
      expect(queryByText("Example 1")).toBeNull();
    });
  });
  it.skip("renders a set of tabs which filter by People type", () => {
    const permissionsWithTypes = [
      {
        acl: createAccessMap(true, true, false, false),
        webId: "https://example1.com/profile/card#me",
        type: "agent",
      },
      {
        acl: createAccessMap(true, true, false, false),
        webId: "https://example-group.com/profile/card#me",
        type: "group",
      },
    ];

    mockedUsePolicyPermissions.mockReturnValue({
      data: permissionsWithTypes,
      mutate: jest.fn(),
    });
    mockedUsePermissionsWithProfiles.mockReturnValue({
      permissionsWithProfiles: permissionsWithProfilesMixedTypes,
    });

    const type = "editors";
    const { getByTestId, queryByText } = renderWithTheme(
      <AgentAccessTable type={type} />
    );

    waitFor(() => {
      // FIXME: change these test ids
      const tabPeople = getByTestId("tab-people");
      const tabGroups = getByTestId("tab-groups");
      userEvent.click(tabPeople);
      expect(queryByText("No people found")).not.toBeNull();
      expect(queryByText("Not a person")).toBeNull();
      userEvent.click(tabGroups);
      expect(queryByText("Example 1")).not.toBeNull();
      expect(queryByText("Example 2")).not.toBeNull();
    });
  });
});
describe("AgentAccessTable without agents", () => {
  it("renders an empty list of permissions if there are no permissions and the policy is not custom", () => {
    const type = "editors";
    mockedUsePolicyPermissions.mockReturnValue({
      data: [],
      mutate: jest.fn(),
    });
    mockedUsePermissionsWithProfiles.mockReturnValue({
      permissionsWithProfiles: [],
    });
    const { asFragment } = renderWithTheme(<AgentAccessTable type={type} />);

    expect(asFragment()).toMatchSnapshot();
  });
  it("renders an empty list of permissions if permissions are unavailable", () => {
    const type = "editors";
    mockedUsePolicyPermissions.mockReturnValue({
      error: "error",
      data: undefined,
      mutate: jest.fn(),
    });
    const { asFragment } = renderWithTheme(<AgentAccessTable type={type} />);

    expect(asFragment()).toMatchSnapshot();
  });
  it("does not render table at all if there aren't any permissions for a custom policy", () => {
    const type = "viewAndAdd";
    mockedUsePolicyPermissions.mockReturnValue({
      data: [],
      mutate: jest.fn(),
    });
    const { asFragment, queryByTestId } = renderWithTheme(
      <AgentAccessTable type={type} />
    );

    expect(asFragment()).toMatchSnapshot();
    expect(queryByTestId(TESTCAFE_ID_AGENT_ACCESS_TABLE)).toBeNull();
  });
});
