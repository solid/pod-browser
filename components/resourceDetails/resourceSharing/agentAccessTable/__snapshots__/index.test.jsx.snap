// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AgentAccessTable with agents renders a list of permissions 1`] = `
<DocumentFragment>
  <div
    class="PodBrowser-root PodBrowser-root PodBrowser-accordion PodBrowser-expanded PodBrowser-rounded PodBrowser-rounded PodBrowser-elevation1 PodBrowser-rounded"
    data-testid="agent-access-table"
  >
    <div
      class="PodBrowser-headerContainer"
    >
      <i
        class="PodBrowser-icon-editor PodBrowser-icon PodBrowser-iconEditor"
      />
      <div
        class="PodBrowser-textContainer"
      >
        <div
          class="PodBrowser-titleAndButtonContainer"
        >
          <p
            class="PodBrowser-title"
          >
            Editors
          </p>
          <button
            class="PodBrowser-button PodBrowser-button--text"
            data-testid="add-agent-button"
          >
            <span
              class="PodBrowser-icon-edit PodBrowser-button__icon PodBrowser-button__icon--before"
            />
            <span
              class="PodBrowser-button__text"
            >
              Edit Editors
            </span>
          </button>
          <button
            class="PodBrowser-button PodBrowser-button--icon-only PodBrowser-action-button"
          >
            <span
              class="PodBrowser-button__text"
            >
              <span
                aria-label="Show menu for policy"
                class="PodBrowser-icon-more"
              />
            </span>
          </button>
        </div>
        <span
          class="PodBrowser-description"
        >
          <p>
            <b>
              Can 
            </b>
            view, edit and delete this resource
          </p>
        </span>
      </div>
    </div>
    <div
      class="PodBrowser-container PodBrowser-entered"
      style="min-height: 0px;"
    >
      <div
        class="PodBrowser-wrapper"
      >
        <div
          class="PodBrowser-wrapperInner"
        >
          <div
            role="region"
          >
            <div
              class="PodBrowser-permissionsContainer"
            >
              <div
                class="PodBrowser-searchBoxContainer"
              >
                <button
                  aria-label="search"
                  class="PodBrowser-root PodBrowser-root"
                  tabindex="0"
                  type="submit"
                >
                  <span
                    class="PodBrowser-label"
                  >
                    <i
                      aria-label="search"
                      class="PodBrowser-icon-search PodBrowser-iconSearch"
                    />
                  </span>
                  <span
                    class="PodBrowser-root"
                  />
                </button>
                <div
                  class="PodBrowser-root PodBrowser-searchInput"
                >
                  <input
                    aria-label="Search by name or WebId"
                    class="PodBrowser-input"
                    data-testid="search-input"
                    placeholder="Search by name or WebId"
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <table
                class="PodBrowser-table PodBrowser-agents-table"
                role="table"
              >
                <tbody
                  class="PodBrowser-table__body"
                  role="rowgroup"
                >
                  <tr
                    class="PodBrowser-table__body-row"
                  >
                    <td
                      class="PodBrowser-table__body-cell PodBrowser-agent-cell"
                    >
                      <div
                        class="PodBrowser-nameAndAvatarContainer"
                      >
                        <div
                          class="PodBrowser-root PodBrowser-circle PodBrowser-avatar PodBrowser-colorDefault"
                        >
                          <span
                            class="PodBrowser-icon-user PodBrowser-avatar__icon"
                          />
                        </div>
                        <p
                          class="PodBrowser-root PodBrowser-detailText PodBrowser-body1 PodBrowser-detailText"
                          data-testid="agent-web-id"
                        >
                          Example A
                        </p>
                        <button
                          class="PodBrowser-button"
                          data-testid="menu-button"
                          type="button"
                        >
                          <i
                            class="PodBrowser-icon-more"
                          />
                        </button>
                      </div>
                    </td>
                  </tr>
                  <tr
                    class="PodBrowser-table__body-row"
                  >
                    <td
                      class="PodBrowser-table__body-cell PodBrowser-agent-cell"
                    >
                      <div
                        class="PodBrowser-nameAndAvatarContainer"
                      >
                        <div
                          class="PodBrowser-root PodBrowser-circle PodBrowser-avatar PodBrowser-colorDefault"
                        >
                          <span
                            class="PodBrowser-icon-user PodBrowser-avatar__icon"
                          />
                        </div>
                        <p
                          class="PodBrowser-root PodBrowser-detailText PodBrowser-body1 PodBrowser-detailText"
                          data-testid="agent-web-id"
                        >
                          Example A
                        </p>
                        <button
                          class="PodBrowser-button"
                          data-testid="menu-button"
                          type="button"
                        >
                          <i
                            class="PodBrowser-icon-more"
                          />
                        </button>
                      </div>
                    </td>
                  </tr>
                  <tr
                    class="PodBrowser-table__body-row"
                  >
                    <td
                      class="PodBrowser-table__body-cell PodBrowser-agent-cell"
                    >
                      <div
                        class="PodBrowser-nameAndAvatarContainer"
                      >
                        <div
                          class="PodBrowser-root PodBrowser-circle PodBrowser-avatar PodBrowser-colorDefault"
                        >
                          <span
                            class="PodBrowser-icon-user PodBrowser-avatar__icon"
                          />
                        </div>
                        <p
                          class="PodBrowser-root PodBrowser-detailText PodBrowser-body1 PodBrowser-detailText"
                          data-testid="agent-web-id"
                        >
                          Example A
                        </p>
                        <button
                          class="PodBrowser-button"
                          data-testid="menu-button"
                          type="button"
                        >
                          <i
                            class="PodBrowser-icon-more"
                          />
                        </button>
                      </div>
                    </td>
                  </tr>
                </tbody>
              </table>
              <div
                class="PodBrowser-showAllButtonContainer"
              >
                <button
                  class="PodBrowser-showAllButton"
                  data-testid="show-all-button"
                  type="button"
                >
                  <p
                    class="PodBrowser-root PodBrowser-body1 PodBrowser-showAllText"
                  >
                    <span>
                      Show all (4) 
                      <i
                        class="PodBrowser-icon-caret-down PodBrowser-showAllButtonIcon"
                      />
                    </span>
                  </p>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`AgentAccessTable without agents does not render table at all if there aren't any permissions for a custom policy 1`] = `<DocumentFragment />`;

exports[`AgentAccessTable without agents renders an empty list of permissions if permissions are unavailable 1`] = `
<DocumentFragment>
  <div
    class="PodBrowser-root PodBrowser-root PodBrowser-accordion PodBrowser-expanded PodBrowser-rounded PodBrowser-rounded PodBrowser-elevation1 PodBrowser-rounded"
    data-testid="agent-access-table"
  >
    <div
      class="PodBrowser-headerContainer"
    >
      <i
        class="PodBrowser-icon-editor PodBrowser-icon PodBrowser-iconEditor"
      />
      <div
        class="PodBrowser-textContainer"
      >
        <div
          class="PodBrowser-titleAndButtonContainer"
        >
          <p
            class="PodBrowser-title"
          >
            Editors
          </p>
          <button
            class="PodBrowser-button PodBrowser-button--text"
            data-testid="add-agent-button"
          >
            <span
              class="PodBrowser-icon-edit PodBrowser-button__icon PodBrowser-button__icon--before"
            />
            <span
              class="PodBrowser-button__text"
            >
              Edit Editors
            </span>
          </button>
          <button
            class="PodBrowser-button PodBrowser-button--icon-only PodBrowser-action-button"
          >
            <span
              class="PodBrowser-button__text"
            >
              <span
                aria-label="Show menu for policy"
                class="PodBrowser-icon-more"
              />
            </span>
          </button>
        </div>
        <span
          class="PodBrowser-description"
        >
          <p>
            <b>
              Can 
            </b>
            view, edit and delete this resource
          </p>
        </span>
      </div>
    </div>
    <div
      class="PodBrowser-container PodBrowser-entered"
      style="min-height: 0px;"
    >
      <div
        class="PodBrowser-wrapper"
      >
        <div
          class="PodBrowser-wrapperInner"
        >
          <div
            role="region"
          >
            <div
              class="PodBrowser-permissionsContainer"
            >
              <span
                class="PodBrowser-emptyStateTextContainer"
              >
                <p>
                  No editors
                </p>
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`AgentAccessTable without agents renders an empty list of permissions if there are no permissions and the policy is not custom 1`] = `
<DocumentFragment>
  <div
    class="PodBrowser-root PodBrowser-root PodBrowser-accordion PodBrowser-expanded PodBrowser-rounded PodBrowser-rounded PodBrowser-elevation1 PodBrowser-rounded"
    data-testid="agent-access-table"
  >
    <div
      class="PodBrowser-headerContainer"
    >
      <i
        class="PodBrowser-icon-editor PodBrowser-icon PodBrowser-iconEditor"
      />
      <div
        class="PodBrowser-textContainer"
      >
        <div
          class="PodBrowser-titleAndButtonContainer"
        >
          <p
            class="PodBrowser-title"
          >
            Editors
          </p>
          <button
            class="PodBrowser-button PodBrowser-button--text"
            data-testid="add-agent-button"
          >
            <span
              class="PodBrowser-icon-edit PodBrowser-button__icon PodBrowser-button__icon--before"
            />
            <span
              class="PodBrowser-button__text"
            >
              Edit Editors
            </span>
          </button>
          <button
            class="PodBrowser-button PodBrowser-button--icon-only PodBrowser-action-button"
          >
            <span
              class="PodBrowser-button__text"
            >
              <span
                aria-label="Show menu for policy"
                class="PodBrowser-icon-more"
              />
            </span>
          </button>
        </div>
        <span
          class="PodBrowser-description"
        >
          <p>
            <b>
              Can 
            </b>
            view, edit and delete this resource
          </p>
        </span>
      </div>
    </div>
    <div
      class="PodBrowser-container PodBrowser-entered"
      style="min-height: 0px;"
    >
      <div
        class="PodBrowser-wrapper"
      >
        <div
          class="PodBrowser-wrapperInner"
        >
          <div
            role="region"
          >
            <div
              class="PodBrowser-permissionsContainer"
            >
              <span
                class="PodBrowser-emptyStateTextContainer"
              >
                <p>
                  No editors
                </p>
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
